### ISSUE TITLE
Model base types are not included even when they are not abstract

### ISSUE THREAD

**Issue by sebastianamaro** [ExRam/ExRam.Gremlinq#1627]
**Describe the bug**
I have a solution which uses GremlinQ. Since I am using a Microservice architecture, I need some common logic between the MS that access the graph, which I put in a shared custom nuget package.
After I did this 2 things started to happen

1) Only on AddV (not on update), i get an Argument Exception saying my object is not part of the model.
2) The Path() response, changed, and instead of having the actual objects in the Objects field of the Path object, i has some Newtonsoft Dynamic representation (I dont have an example now, but if you need it I could try to recreate this).

**Expected behavior**
The only thing I had in this Nuget package for now, is the Vertex and Edge classes, since they are for sure common to all the different MS.
I expect that the library can identify that the type exists, and allow me to abstract some pieces of code into this Nuget package.

**Version information**
12.7.0

**Minimal working example**
https://github.com/sebastianamaro/Gremlinqissue

I included there the Nuget package Project, so you can set that up in your enviroment. As you see it only contains 3 classes. 

**Additional context**
I think the issue relies here. GraphElementModel.cs Line 130
```
public ElementMetadata? TryGetMetadata(Type elementType) => _elementMetadataOverrides.TryGetValue(elementType, out var elementMetadata)
    ? elementMetadata
    : IsWithinModel(elementType)
        ? new ElementMetadata(elementType.Name)
        : default(ElementMetadata?);

private bool IsWithinModel(Type type) => typeof(TBaseType).IsAssignableFrom(type) && _assemblies.Contains(type.Assembly);

```
        
        
I debugged it, and _assemblies only contains the "Shared" assembly. And type.Assembly is my solutions assembly, 
typeof(TBaseType).IsAssignableFrom(type) returns true.

**Comment by sebastianamaro** [comment]
Nevermind, I found the solution. I leave it here for other people.

`var _g = g
    .UseCosmosDb<Vertex, Edge>(configurator => configurator
        .At(new Uri("URI"))
        .OnDatabase("DB")
        .OnGraph("GRAPH")
        .WithPartitionKey(x => x.TenantId!)
        .AuthenticateBy("Key")
        .UseNewtonsoftJson())
    .ConfigureEnvironment(env=> env.UseModel(GraphModel.FromBaseTypes<Vertex, Edge>().AddAssemblies(typeof(Node).Assembly)));`
    
    
You have to configure the GraphModel, and add the Assembly of the Non-Nuget project (since it automatically adds the assemble where the Vertex and Edge classes are located)

**Comment by sebastianamaro** [comment]
By the way, it would be REALLY helpful, if there is some kind of documentation about the possible configurations. 
If not, at least i think a link to the most relevant Unit Tests that prove this might be good, that is how I found the solution.

**Comment by sebastianamaro** [comment]
I found another issue related to this

I have this query:

 var paths = await _g.V<Scenario>(scenarioId)
     .Where(x => x.TenantId == tenantId)
     .As("scenario")
     .Cast<object>()
     .Out<Includes>()
     .As("start")
     .Cast<Vertex>()
     .OutE<CanInteractWith>()
     .As("interaction")
     .InV()
     .Cast<Vertex>()
     .As("end")
     .Path().ToArrayAsync();
     
Vertex is defined in the Nuget package, while Scenario and CanInteractWith are defined in this project.

For some reason I dont know, this path query in debug is resolved to this:
g.V('d8503822-1d91-454b-97be-600b9cb8d720').hasLabel('Scenario').has('TenantId','123').as('scenario').out('Includes').as('start').outE('CanInteractWith').as('interaction').inV().hasLabel('Scenario').as('end').path()

As you can see, the inV() is filtering by label Scenario, but my query does not state that, it states it should be a Vertex. 

For know I resolved it using the non generic version of InV(), since I do not do any additional filtering.

**Comment by danielcweber** [comment]
@sebastianamaro Feel free to extract the latter issue into a separate issue with a minimal repro.

**Comment by sebastianamaro** [comment]
https://github.com/sebastianamaro/Gremlinqissue

Here is again in the latest commit the repro

**Comment by danielcweber** [comment]
In any case, use the `.AddAssemblies(typeof(Node).Assembly))` part, it's needed if Node lives in another assembly There is indeed a bug, however.

**Comment by danielcweber** [comment]
Though there is a bug, I'd recommend marking the base vertex/edge-types as abstract and never creating them directly but only derived types. There is not much information in an "Edge" label alone and this might already be a workaround.

**Comment by danielcweber** [comment]
Fixed by #1651.
